src/main/java/bio/terra/app/controller/AdminApiController.java

```diff
@@ -113,15 +114,32 @@ public ResponseEntity<SnapshotModel> adminRetrieveSnapshot(UUID id) {
             SnapshotRetrieveIncludeModel.CREATION_INFORMATION,
             SnapshotRetrieveIncludeModel.DUOS);
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
-    logger.info(
-        "Verifying resource type admin authorization: {} for resource type: {} and resource id: {}",
-        userReq.getEmail(),
-        IamResourceType.DATASNAPSHOT,
-        id);
+    logger.info(AUTH_DESCRIPTION, userReq.getEmail(), IamResourceType.DATASNAPSHOT, id);
     iamService.verifyResourceTypeAdminAuthorized(
         userReq, IamResourceType.DATASNAPSHOT, IamAction.ADMIN_READ_SUMMARY_INFORMATION);
     logger.info("Retrieving snapshot id: {}", id);
     SnapshotModel snapshotModel = snapshotService.retrieveSnapshotModel(id, include, userReq);
     return ResponseEntity.ok(snapshotModel);
   }
+
+  @Override
+  public ResponseEntity<JobModel> adminInheritSteward(UUID id, Boolean inheritSteward) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    logger.info(AUTH_DESCRIPTION, userReq.getEmail(), IamResourceType.DATASET, id);
+    iamService.verifyResourceTypeAdminAuthorized(
+        userReq, IamResourceType.DATASET, IamAction.ADMIN_TOGGLE_INHERIT_STEWARD);
+
+    // dataset already has the requested value for inheritSteward
+    if (datasetService.retrieveDatasetSummary(id).isInheritSteward().equals(inheritSteward)) {
+      throw new BadRequestException(
+          String.format("Dataset %s already has inheritSteward set to %s", id, inheritSteward));
+    }
+
+    if (inheritSteward) {
+      String jobId = datasetService.enableInheritSteward(id, userReq);
+      return jobToResponse(jobService.retrieveJob(jobId, userReq));
+    } else {
+      throw new NotImplementedException("disabling Inherit Steward is not implemented yet.");
+    }
```

Comment:
No need to use `else` after a `return`
```suggestion
    }
    throw new NotImplementedException("disabling Inherit Steward is not implemented yet.");
```